File: WooCommerce/src/main/kotlin/com/woocommerce/android/viewmodel/ViewModelKey.java
package com.woocommerce.android.viewmodel;

import java.lang.annotation.*;

import androidx.lifecycle.ViewModel;
import dagger.MapKey;

@MapKey
@Documented
@Target({ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
public @interface ViewModelKey {
    Class<? extends ViewModel> value();
}

File: WooCommerce/src/main/kotlin/com/woocommerce/android/viewmodel/ViewModelKey.kt
package com.woocommerce.android.viewmodel

import androidx.lifecycle.ViewModel

import dagger.MapKey
import kotlin.annotation.AnnotationRetention
import kotlin.annotation.AnnotationTarget
import kotlin.annotation.MustBeDocumented
import kotlin.reflect.KClass

@MustBeDocumented
@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
@MapKey
annotation class ViewModelKey(val value: KClass<out ViewModel>)
